app/updateSNAPP.ts

```diff
@@ -0,0 +1,135 @@
+import path from 'path';
+import { app, remote } from 'electron';
+import { spawn, exec, execFile } from 'child_process';
+import psList from 'ps-list';
+import { logger } from '$Logger';
+
+const { platform } = process;
+const MAC_OS = 'darwin';
+const LINUX = 'linux';
+const WINDOWS = 'win32';
+
+const getAppInstallDir = () => {
+    const homeDirectory = app ? app.getPath( 'home' ) : remote.app.getPath( 'home' );
+    let installTargetDirectory = path.resolve( '/Applications' );
+
+    if ( platform === LINUX ) {
+        installTargetDirectory = path.resolve( homeDirectory, 'bin' );
+    }
+    if ( platform === WINDOWS ) {
+    //  ~/AppData/Local/Programs/safe-launch-pad/safe Launch Pad.exe
+        installTargetDirectory = path.resolve(
+            homeDirectory,
+            'AppData',
+            'Local',
+            'Programs'
+        );
+    }
+    return installTargetDirectory;
+};
+
+const getApplicationExecutable = ( application ): string => {
+    // https://github.com/joshuef/electron-typescript-react-boilerplate/releases/tag/v0.1.0
+    // TODO ensure name conformity with download, or if different, note how.
+
+    let applicationExecutable: string;
+
+    switch ( platform ) {
+        case MAC_OS: {
+            applicationExecutable = `${application.name ||
+        application.packageName}.app`;
+            break;
+        }
+        case WINDOWS: {
+            applicationExecutable = path.join(
+                `${application.packageName || application.name}`,
+                `${application.name || application.packageName}.exe`
+            );
+            break;
+        }
+        case LINUX: {
+            applicationExecutable = `${application.packageName ||
+        application.name}.AppImage`;
+            break;
+            // electron-react-boilerplate-0.1.0-x86_64.AppImage
+        }
+        default: {
+            logger.error( 'Unsupported platform for desktop applications:', platform );
+        }
+    }
+    return applicationExecutable;
+};
+
+const getInstalledLocation = ( application ): string => {
+    const applicationExecutable = getApplicationExecutable( application );
+    const installedPath = path.join( getAppInstallDir(), applicationExecutable );
+
+    logger.error( 'installedPath', installedPath );
```

Comment:
error?

Reply:
Yep, that has to be changed. I was using that to make certain logs more distinguishable

Reply:
You can filter logs for a given string in devtools. Ideal is to get decent log messages (at appropriate levels), from the get go so we dont accidentally commit such things. (it's a lofty goal. I miss nonsense logs all the time myself :P )